facade/src/main/java/org/jboss/pnc/facade/providers/ProductMilestoneProviderImpl.java

```diff
@@ -184,4 +214,96 @@ public void cancelMilestoneCloseProcess(String id)
 
         return queryForCollection(pageIndex, pageSize, sortingRsql, query, withProductVersionId(Integer.valueOf(productVersionId)));
     }
+
+    @Override
+    public Page<MilestoneInfo> getMilestonesOfArtifact(String id, int pageIndex, int pageSize) {
+        CriteriaBuilder cb = em.getCriteriaBuilder();
+
+        List<Integer> buildIn = getBuildIds(cb, id, Artifact_.buildRecords);
+        List<Integer> dependencyOf = getBuildIds(cb, id, Artifact_.dependantBuildRecords);
+
+        Set<Integer> milestoneIds = new HashSet<>(dependencyOf);
+        milestoneIds.addAll(buildIn);
+        milestoneIds.remove(null); // some builds are not in a milestone and so it gives us null
+        if(milestoneIds.isEmpty()){
+            return new Page<>();
+        }
+
+        CriteriaQuery<Tuple> query = milestoneInfoQuery(cb, milestoneIds);
+        int offset = pageIndex * pageSize;
+        List<MilestoneInfo> milestones = em.createQuery(query)
+                .setMaxResults(pageSize)
```

Comment:
Is it possible to use LIMIT in DB query?

Reply:
setMaxResults is not setting LIMIT in the DB query?